src/vs/workbench/api/common/extHostWorkspace.ts

```diff
@@ -573,6 +576,37 @@ export class ExtHostWorkspace implements ExtHostWorkspaceShape, IExtHostWorkspac
 			this._onDidGrantWorkspaceTrust.fire();
 		}
 	}
+
+	// --- edit sessions ---
+
+	// called by ext host
+	registerEditSessionIdentityProvider(scheme: string, provider: vscode.EditSessionIdentityProvider) {
+		this._EditSessionIdentityProviders[scheme] = provider;
+		this._proxy.$registerEditSessionIdentityProvider(scheme);
+		return toDisposable(() => {
+			delete this._EditSessionIdentityProviders[scheme];
```

Comment:
Does this lack an `$unregisterEditSession...` invocation/call? How else does the renderer/core know that a provider is gone